internal/validator/cosignvalidator/cosign_validator.go

```diff
@@ -35,10 +33,16 @@ type CosignValidator struct {
 	Name string `validate:"required"`
 	// type of the validator (will always be "cosign")
 	Type string `validate:"eq=cosign"`
-	// url of the Rekor instance
-	Rekor string `validate:"url"`
+	// Rekor instance
+	Rekor *client.Rekor
+	// public key of the Rekor instance or nil if using the sigstore one
+	rekorPubkey *cosign.TrustedTransparencyLogPubKeys
+	// certificate of the fulcio instance or nil if using the sigstore one
+	fulcioCert *x509.CertPool
+	// public key of the CT log or nil if using the sigstore one
```

Comment:
nit
```suggestion
	// public key of the Rekor instance or nil if using the Sigstore one
	rekorPubkey *cosign.TrustedTransparencyLogPubKeys
	// certificate of the Fulcio instance or nil if using the Sigstore one
	fulcioCert *x509.CertPool
	// public key of the CT log or nil if using the Sigstore one
```
or lower-case Rekor, but this way it's inconsistent